bindings/src/test/java/io/pravega/storage/extendeds3/S3Mock.java

```diff
@@ -0,0 +1,315 @@
+            }
+
+            od.content = BufferView.wrap(Arrays.asList(objectContent, new ByteArraySegment(buffer)));
+        }
+    }
+
+    public void setObjectAcl(String bucketName, String key, AccessControlList acl) {
+        String objectName = getObjectName(bucketName, key);
+        synchronized (this.objects) {
+            ObjectData od = this.objects.get(objectName);
+            if (od == null) {
+                throw new S3Exception("NoObject", 404, "NoSuchKey", key);
+            }
+            od.acl = acl;
+        }
+    }
+
+    public void setObjectAcl(SetObjectAclRequest request) {
+        String objectName = getObjectName(request.getBucketName(), request.getKey());
+        synchronized (this.objects) {
+            ObjectData od = this.objects.get(objectName);
+            if (od == null) {
+                throw new S3Exception("NoObject", 404, "NoSuchKey", request.getKey());
+            }
+            od.acl = request.getAcl();
+        }
+    }
+
+    public AccessControlList getObjectAcl(String bucketName, String key) {
+        String objectName = getObjectName(bucketName, key);
+        synchronized (this.objects) {
+            ObjectData od = this.objects.get(objectName);
+            if (od == null) {
+                throw new S3Exception("NoObject", 404, "NoSuchKey", key);
+            }
+
+            return od.acl;
+        }
+    }
+
+    public CopyPartResult copyPart(CopyPartRequest request) {
+        String objectName = getObjectName(request.getBucketName(), request.getKey());
+        synchronized (this.objects) {
+            Map<Integer, CopyPartRequest> partMap = this.multipartUploads.get(objectName);
+            if (partMap == null) {
+                throw new S3Exception("NoSuchKey", 404, "NoSuchKey", "");
+            }
+            partMap.put(request.getPartNumber(), request);
+            CopyPartResult result = new CopyPartResult();
+            result.setPartNumber(request.getPartNumber());
+            result.setETag(request.getUploadId());
+            return result;
+        }
+    }
+
+    public void deleteObject(String bucketName, String key) {
+        String objectName = getObjectName(bucketName, key);
+        synchronized (this.objects) {
+            if (this.objects.remove(objectName) == null) {
+                throw new S3Exception("NoSuchKey", 404, "NoSuchKey", "");
+            }
+        }
+    }
+
+    public DeleteObjectsResult deleteObjects(DeleteObjectsRequest request) {
+        for (ObjectKey obj : request.getDeleteObjects().getKeys()) {
+            deleteObject(request.getBucketName(), obj.getKey());
+        }
+        return new DeleteObjectsResult();
+    }
+
+    public ListObjectsResult listObjects(String bucketName, String prefix) {
+        ListObjectsResult result = new ListObjectsResult();
+        List<S3Object> list;
+        String objectPrefix = getObjectName(bucketName, prefix);
+        String bucketPrefix = getObjectName(bucketName, "");
```

Comment:
Seems bit hacky doing it this way.
may be use a special separator char eg.  `:` --> `bucket_name:object_name` 

or simply use bucketName.length+1

Reply:
Fixed.